crates/sui-core/src/authority/test_authority_builder.rs

```diff
@@ -165,9 +165,12 @@ impl<'a> TestAuthorityBuilder<'a> {
         let registry = Registry::new();
         let cache_metrics = Arc::new(ResolverMetrics::new(&registry));
         let signature_verifier_metrics = SignatureVerifierMetrics::new(&registry);
+        // `_guard` must be declared here so it is not dropped before
+        // `AuthorityPerEpochStore::new` is called
+        let _guard;
         if self.protocol_config.is_some() {
             let config = self.protocol_config.unwrap();
-            let _guard = ProtocolConfig::apply_overrides_for_testing(move |_, _| config.clone());
+            _guard = ProtocolConfig::apply_overrides_for_testing(move |_, _| config.clone());
         }
```

Comment:
```rust
let guard = self.protocol_config.cloned().map(|c| {
    ProtocolConfigConfig::apply_overrides_for_testing(move |_, _| c)
});
```

Reply:
I could not get the clone to go away and have zero understanding as to why (Rust lambdas man...)